File: Duomenų struktūrų naudojimas/Python/Deque_(deklas).py
from collections import deque

# Sukuriame tuščią deklą
d = deque()

# Pridedame elementus į galą ir į priekį
d.append(10)       # Prideda į galą
d.appendleft(20)   # Prideda į priekį

# Pašaliname elementus iš galo ir priekio
d.pop()            # Pašalina iš galo
d.popleft()        # Pašalina iš priekio

print(d)  # Išveda deque([])

File: Duomenų struktūrų naudojimas/C++/Deque_(deklas).cpp
#include <iostream>
#include <deque>

int main() {
    std::deque<int> d;

    // Pridedame elementus į galą ir į priekį
    d.push_back(10);    // Prideda į galą
    d.push_front(20);   // Prideda į priekį

    // Pašaliname elementus iš galo ir priekio
    d.pop_back();       // Pašalina iš galo
    d.pop_front();      // Pašalina iš priekio

    // Patikriname, ar deklas tuščias
    if (d.empty()) {
        std::cout << "Deklas yra tuščias." << std::endl;
    }

    return 0;
}